Automattic/wp-calypso
# Editor: Move post_tag usage to terms

This branch seeks to move the tracking of `post.tags` to redux and the `post.terms` object.  This is the final step that allows us to remove the legacy flux terms store, and related tags code.

The majority of the key changes can be seen in 49056b7 where post tags are moved over to utilize `<TermTokenField />`.  f678fc5 is cherry-picked from #7937 and fixes an issue where `post_tags` for the `page` post type where being cached in the local state tree incorrectly.  While this branch is not blocked by that PR - testing without this bug was easier for me.  And in 1d74481 the ✂️ were deployed to remove all the old terms flux implementation.

Also worth noting if you happen to test out page `post_tags`, there is a bug in the API currently where `post_tags` are not being populated in the `terms` object for pages.  Fix pending at D2769-code.

__To Test__
- Ensure the test suites pass, which circle ci kindly does for us already
- Interact with the post editor, specifically the post tags token field in a variety of ways.  Add new tags, use existing tags, modify tags on published posts, edit tags and attempt to navigate away... generally just try to bend it till it breaks.

Test live: https://calypso.live/?branch=update/editor/post-tags-to-term-token-field
